compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSTypeImpl.kt

```diff
@@ -60,7 +60,7 @@ class KSTypeImpl private constructor(
     }
 
     override val arguments: List<KSTypeArgument> by lazy {
-        kotlinType.arguments.map { KSTypeArgumentDescriptorImpl.getCached(it) }
+        kotlinType.arguments.map { KSTypeArgumentDescriptorImpl.getCached(it, Origin.SYNTHETIC) }
```

Comment:
I'm not sure if SYNTHETIC is the best way to describe origin here. IMO there is no answer for origin in KSType.